src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -802,6 +804,50 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// 210824 Touch of the Magi (Aura) - SPELL_MAGE_TOUCH_OF_THE_MAGI_AURA
+class spell_mage_touch_of_the_magi_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_touch_of_the_magi_aura);
+
+    uint32 _sum = 0;
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_TOUCH_OF_THE_MAGI_EXPLODE });
+    }
+
+    void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        DamageInfo* damageInfo = eventInfo.GetDamageInfo();
+        if (damageInfo)
+            if (damageInfo->GetAttacker() == GetCaster() && damageInfo->GetVictim() == GetTarget())
+            {
+                uint32 extra = CalculatePct(damageInfo->GetDamage(), 25);
+                if (extra > 0)
+                {
+                    _sum += extra;
+                    AuraEffect* auraEffect = GetTarget()->GetAuraEffect(SPELL_MAGE_TOUCH_OF_THE_MAGI_AURA, 0, GetCaster()->GetGUID());
+                    auraEffect->ChangeAmount(_sum);
+                }
+            }
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (_sum == 0 || GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
```

Comment:
`_sum` in this function is the same as `aurEff->GetAmount()`

Reply:
so we can drop `_sum` completely, pog